packages/react-native-web/src/exports/Modal/ModalFocusTrap.js

```diff
@@ -0,0 +1,175 @@
+  } catch (e) {
+    // Do nothing
+  }
+
+  return getActiveElement() === element;
+}
+
+function focusFirstDescendant(element: any) {
+  for (let i = 0; i < element.childNodes.length; i++) {
+    const child = element.childNodes[i];
+    if (attemptFocus(child) || focusFirstDescendant(child)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+function focusLastDescendant(element: any) {
+  for (let i = element.childNodes.length - 1; i >= 0; i--) {
+    const child = element.childNodes[i];
+    if (attemptFocus(child) || focusLastDescendant(child)) {
+      return true;
+    }
+  }
+  return false;
+}
+
+export type ModalFocusTrapProps = {|
+  children?: any,
+
+  active?: boolean | () => boolean,
+|};
+
+const ModalFocusTrap = ({ active, children }: ModalFocusTrapProps) => {
+  const trapElementRef = useRef<?Node>();
+
+  // Ref used to track trapping of focus and to prevent focus from leaving a modal
+  // for accessibility reasons per W3CAG.
+  const focusRef = useRef<{ trapFocusInProgress: boolean, lastFocusedElement: ?HTMLElement }>({
+    trapFocusInProgress: false,
+    lastFocusedElement: null
+  });
+
+  const trapFocus = useCallback(() => {
+    let activeElement = getActiveElement();
+
+    // We should not trap focus if:
+    // - The modal hasn't fully initialized with an HTMLElement ref
+    // - Focus is already in the process of being trapped (eg, we're refocusing)
+    // - isTrapActive prop being false-ish tells us to do nothing
+    if (!trapElementRef.current || focusRef.current.trapFocusInProgress || !active) {
+      return;
+    }
+
+    try {
+      focusRef.current.trapFocusInProgress = true;
+
+      // Only muck with the focus if the event target isn't within this modal
+      if (activeElement instanceof Node && !trapElementRef.current.contains(activeElement)) {
+        // To handle keyboard focusing we can make an assumption here.
+        // If you're tabbing through the focusable elements, the previously
+        // active element will either be the first or the last.
+        //
+        // If the previously selected element is the "first" descendant
+        // and we're leaving it - this means that we should
+        // be looping around to the other side of the modal.
+        let hasFocused = focusFirstDescendant(trapElementRef.current);
+        if (focusRef.current.lastFocusedElement === activeElement) {
+          hasFocused = focusLastDescendant(trapElementRef.current);
+        }
+
+        // Reload fetching the active element
+        activeElement = getActiveElement();
+
+        // If we couldn't focus a new element then we need to blur the active element
+        if (!hasFocused && activeElement) {
+          activeElement.blur();
+        }
+      }
+    } finally {
+      focusRef.current.trapFocusInProgress = false;
+    }
+
+    focusRef.current.lastFocusedElement = getActiveElement();
+  }, [active]);
+
+  // Bind to the document itself for this component
+  useEffect(() => {
```

Comment:
Move everything in this effect into canUseDOM check and then you can avoid the check in all the other functions. This will also need to use the "isomorphic" useLayoutEffect to attach these handlers before paint, which I'll add when committing this